implementation.js

```diff
@@ -1,8 +1,43 @@
 'use strict';
 var ES = require('es-abstract/es6');
 var supportsDescriptors = require('define-properties').supportsDescriptors;
+var global = require('system.global')();
+var isCallable = require('is-callable');
+var iterator = typeof Symbol === 'function' ? Symbol.iterator : '@@iterator';
+var isIterable = function (target) {
+	return typeof target[iterator] === 'function';
+};
+var fromIterable = function (iterable) {
+	var array = [];
+	var iter = iterable[iterator]();
+	var done = false;
+	while (!done) {
+		var res = iter.next();
+		var value = res.value;
+		done = res.done;
+		if (!done) {
+			array.push(value);
+		}
+	}
+	return array;
+};
 
 /*! https://mths.be/array-from v0.2.0 by @mathias */
+var parse = function (items) {
+	if (isIterable(items)) {
+		return fromIterable(items);
+	}
+	if (typeof items === 'string') {
+		return items.match(/[\uD800-\uDBFF][\uDC00-\uDFFF]?|[^\uD800-\uDFFF]|./g) || [];
+	}
+	if (global.Map && items instanceof global.Map && isCallable(items.entries)) {
```

Comment:
`instanceof` will not work cross-realm (like iframes etc). We can't rely on anything using `instanceof` of `===`.